launcher_utils/advertise.go

```diff
@@ -103,14 +105,73 @@ func Advertise(ctx context.Context, servers []server_structs.XRootDServer) error
 	return firstErr
 }
 
+// Get the site name from the registry given a namespace prefix
+func getSitenameFromReg(ctx context.Context, prefix string) (sitename string, err error) {
+	fed, err := config.GetFederation(ctx)
+	if err != nil {
+		return
+	}
+	if fed.NamespaceRegistrationEndpoint == "" {
+		err = fmt.Errorf("unable to fetch site name from the registry. Federation.RegistryUrl is unset")
```

Comment:
This error message could be confusing to someone who uses `Federation.DiscoveryUrl` instead of directly setting the registry/director URLs. Can you modify it to take that case into account?